### ISSUE TITLE
Task - Task report avatar in LHN shows task assignee instead of task creator before opening it

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#33461]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.15-4
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
Precondition: User A and B have no prior chat and they are on private domain
1. [User A] In 1:1 DM with User C. create a task and assign it to User B
2. [User B] Note the avatar of task report in LHN
3. [User B] Open the task report

## Expected Result:
In Step 2, when User B receives the task, the task report avatar should display the avatar of task creator

## Actual Result:
In Step 2, when User B receives the task, the task report avatar displays task assignee avatar (User B) instead of task creator avatar
The task report avatar only shows the correct avatar (task creator) after opening the task report in Step 3

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/3c783608-b077-4a8e-9f6a-dad612a45700

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by roryabraham** [comment]
Weird

**Comment by roryabraham** [comment]
Well what I'm seeing is that in step 2, the `ownerAccountID` is user `A` and the `mangagerAccountID` on the report is user B. Now going to click on the report as user B and see what happens.

**Comment by roryabraham** [comment]
Clarification – I was looking at the _task report_, but there's also a _chat report_ that's created between users A and B

**Comment by roryabraham** [comment]
So it looks like the chat report that's optimistically created when assigning the task has the wrong `participantAccountIDs`

**Comment by roryabraham** [comment]
Or rather, not just the one that's optimistically created, but the one that user B receives via Pusher

**Comment by roryabraham** [comment]
Before user B clicks on the report, the `participantAccountIDs` array just had the accountID of B, in this case `[185]`. But _after_ user B clicks on the report, the `participantAccountIDs` array had just the accountID of B, in this case `[186]`.

Furthermore, I don't see any code that optimistically generates the chat report in this case.

**Comment by roryabraham** [comment]
*Edit:* the chat report is optimistically generated [here](https://github.com/Expensify/App/blob/14bc32a3d22066fa4fc18c78245d510e4f75c46d/src/libs/actions/Task.js#L594)

**Comment by roryabraham** [comment]
At this point, I'm pretty sure that this is a back-end issue, so I'm going to remove the deploy blocker label and check this off.

**Comment by roryabraham** [comment]
In short, the problem here is:

- When user A uses `CreateTask` to create a chat report with user B:
    - the `participantAccountIDs` returned by the API for user A is (correctly?) `[$userBAccountID]`
    - the `participantAccountIDs` returned by the API for user B is (incorrectly?) `[$userBAccountID]` (i.e: the `participantAccountIDs` are not tailored for each users' perspective.
- When user B uses `OpenReport` to open that same report, then `participantAccountIDs` is corrected to be `[$userAAccountID]`

**Comment by roryabraham** [comment]
I'm thinking that an approach to solving this might be to leverage `ReportUtils::pushNewChat` in `ReportUtils::getOnyxDataForNewTaskAssignee`. There's some logic in `ReportUtils::pushNewChat` that sends a different value for `participantAccountIDs` to each participant, excluding the recipient. That's what we need to use here.

I also notice that it looks like some of the Onyx post-processing could be DRYed up between `ReportAPI::createTask` and `ReportAPI::editTask`.

Alternatively, we could make the jump to true 1:1:1 and move all the Onyx logic from PHP to Auth for both `CreateTask` and `EditTask`.

**Comment by roryabraham** [comment]
cc @thienlnam for a buddy check

**Comment by thienlnam** [comment]
It seems odd to me that the participantAccountIDs change based on who is opening the report - Is this something we updated recently? I would expect that they would always be the same regardless of who is opening it and then the personalDetails would be different depending on the user

> When user B uses OpenReport to open that same report, then participantAccountIDs is corrected to be [$userAAccountID]

I would check why this is the case, we haven't historically included the report creator in the participants so there might have been a recent change (maybe for UCR?)

I'm sure your approach would work, and maybe the correct solution if it's intended that the participantAccountIDs are now intended to be different per user

**Comment by roryabraham** [comment]
> It seems odd to me that the participantAccountIDs change based on who is opening the report

Good point! That is a bit weird, I'm not sure why it was built that way. I'll look into it a bit more

**Comment by roryabraham** [comment]
No update since OOO

**Comment by roryabraham** [comment]
Asked about the plan for a potentially related issue [here](https://github.com/Expensify/App/issues/33191#issuecomment-1872351652)

**Comment by roryabraham** [comment]
Alright, well this is internal and I have a few other things on my plate so I don't think I'll be able to prioritize this one this week.

The updated plan of action is to:

1. Update the front-end to always expect chat reports to have their full list of participants, including the current user. This will involve many changes in many places in E/App.
1. Update the PHP layer to stop stripping out the current user from `participantAccountIDs`

This will make it simpler to go full 1:1:1 in the future, but I'm not including that in the scope to fix this issue, because it would take way too much work to migrate all places where we update `participantAccountIDs` to perform Onyx updates from Auth.

**Comment by roryabraham** [comment]
Sorry, still no update. Maybe tomorrow, but I think more realistic goal to make progress on this is Wednesday

**Comment by roryabraham** [comment]
Still no update. Going to bump this to weekly for now since I have a few higher priority items ahead of it in my TODO queue

**Comment by roryabraham** [comment]
Still no update unfortunately

**Comment by sonialiap** [comment]
@roryabraham any updates this week? 😁

**Comment by roryabraham** [comment]
nope, sorry have been focused on more urgent wave work and chores.

**Comment by roryabraham** [comment]
Throwing this on HOLD for https://github.com/Expensify/Expensify/issues/341000

**Comment by roryabraham** [comment]
still on HOLD

**Comment by kavimuru** [comment]
Issue is reproducible still.


https://github.com/Expensify/App/assets/43996225/7e15bb08-60f7-45e6-868f-29e74fd6c6d3

**Comment by roryabraham** [comment]
Ok, unfortunate that this is still reproducible. Will need to circle back

**Comment by roryabraham** [comment]
unassigning myself as I'm not sure when I'll be able to prioritize this

**Comment by sonialiap** [comment]
@saracouto this issue was moved to "paused" and it's been open since December, should we close this out?

**Comment by sonialiap** [comment]
Paused

**Comment by sonialiap** [comment]
I'm OOO Aug 19-30, but this is paused so not adding a leave buddy

**Comment by sonialiap** [comment]
Paused

**Comment by sonialiap** [comment]
Asking for a retest

**Comment by sonialiap** [comment]
asked for new retest - [slack](https://expensify.slack.com/archives/C9YU7BX5M/p1731434005087869)

**Comment by mvtglobally** [comment]
Not reproducible 
https://github.com/user-attachments/assets/7cc093ec-842d-4627-8216-9fec86ec5b3a
